src/main/java/pageObjects/Licenses.java

```diff
@@ -0,0 +1,264 @@
+}
+public void createLicenseWithExpirationDateEarlierThanPurchaseDate()
+{
+    signInAsAdmin();
+    licenseButton.click();
+    wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("add-license")));
+    addLicenseButton.click();
+    wait.until(ExpectedConditions.urlToBe(Constants.LICENSE_DETAILS_URL));
+    nameField.click();
+    nameField.sendKeys(LICENSE_NAME_WS);
+    expirationDate.click();
+    expirationDate.sendKeys(Keys.CONTROL + "a");
+    expirationDate.sendKeys(Keys.DELETE);
+    expirationDate.sendKeys((EXPIRATION_DATE_WS));
+    saveLicense.click();
+    wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("error_toaster")));
+}
+    public void createLicenseWithEmptyExpirationDate (){
+        signInAsAdmin();
+        licenseButton.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("add-license")));
+        addLicenseButton.click();
+        wait.until(ExpectedConditions.urlToBe(Constants.LICENSE_DETAILS_URL));
+        nameField.click();
+        nameField.sendKeys(LICENSE_NAME_WS);
+        purchaseDate.click();
+        purchaseDate.sendKeys(Keys.CONTROL + "a");
+        purchaseDate.sendKeys(Keys.DELETE);
+        purchaseDate.sendKeys((PURCHASE_DATE_WS));
+        expirationDate.click();
+        expirationDate.sendKeys(Keys.CONTROL + "a");
+        expirationDate.sendKeys(Keys.DELETE);
+        saveLicense.click();
+    }
+    public void createLicenseWithEmptyPurchaseDate (){
+        signInAsAdmin();
+        licenseButton.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("add-license")));
+        addLicenseButton.click();
+        wait.until(ExpectedConditions.urlToBe(Constants.LICENSE_DETAILS_URL));
+        nameField.click();
+        nameField.sendKeys(LICENSE_NAME_WS);
+        purchaseDate.click();
+        purchaseDate.sendKeys(Keys.CONTROL + "a");
+        purchaseDate.sendKeys(Keys.DELETE);
+        expirationDate.click();
+        expirationDate.sendKeys(Keys.CONTROL + "a");
+        expirationDate.sendKeys(Keys.DELETE);
+        expirationDate.sendKeys((EXPIRATION_DATE_WS));
+        saveLicense.click();
+    }
+    public void successfulEditLicenseName()
+    {
+        signInAsAdmin();
+        licenseButton.click();
+        wait.until(ExpectedConditions.urlToBe(Constants.LICENSES_URL));
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("license_dropdown_11111")));
+        licenceDropdown.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("license_edit_11111")));
+        licenceEditDropdown.click();
+        wait.until(ExpectedConditions.urlToBe(Constants.LICENSE_DETAILS_URL));
+        nameField.click();
+        nameField.clear();
+        nameField.sendKeys(LICENSE_NAME_WS);
+        saveLicense.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("success_toaster")));
+    }
+
+//    public void successfulEditExpirationDate()
```

Comment:
if we dont use this, leave a comment so we will know in the future the reason for commenting this method